extension-workspace-references.md

```diff
@@ -0,0 +1,135 @@
+   */
+  xworkspaceReferencesProvider?: boolean;
+}
+```
+
+#### Workspace References Request
+
+The workspace references request is sent from the client to the server to export project-wide references to dependencies. That is, the response only returns references in the project to symbols defined in dependencies.
+
+_Request_
+* method: 'workspace/xreferences'
+* params: `WorkspaceReferencesParams` defined as follows:
+```typescript
+/**
+ * The parameters of a Workspace References Request.
+ */
+interface WorkspaceReferencesParams {
+}
+```
+
+_Response_:
+* result: `ReferenceInformation[]` defined as follows:
+```typescript
+/**
+ * Represents information about a reference to programming constructs like
+ * variables, classes, interfaces etc.
+ */
+interface ReferenceInformation {
+    /**
+     * The location in the workspace where the `symbol` has been referenced.
+     */
+    reference: Location;
+
+    /**
+     * Metadata information describing the symbol being referenced.
+     */
+    symbol: SymbolMetadata;
+}
+```
+* error: code and message set in case an exception happens during the workspace references request.
+
+Where `SymbolMetadata` is defined as follows:
+
+```typescript
+/**
+ * Represents information about a programming construct like a variable, class,
+ * interface etc that has a reference to it. Effectively, it contains data similar
+ * to SymbolInformation except all fields are optional.
+ *
+ * SymbolMetadata usually uniquely identifies a symbol, but it is
+ * not guaranteed to do so.
+ */
+interface SymbolMetadata {
+    /**
+     * The name of this symbol (same as `SymbolInformation.name`).
+     */
+    name?: string;
+
+    /**
+     * The kind of this symbol (same as `SymbolInformation.kind`).
+     */
+    kind?: number;
+
+    /**
+     * The URI of this symbol (same as `SymbolInformation.location.uri`).
+     */
+    uri?: string;
+
+    /**
+     * The name of the symbol containing this symbol (same as `SymbolInformation.containerName`).
+     */
+    containerName?: string;
+
+    /**
+     * The repository URI that the symbol is defined in, e.g. `github.com/golang/go`
+     * or `bitbucket.org/jespern/django-piston`.
+     */
+    repo?: string;
+
+    /**
+     * The 'package', 'library', or 'crate' name that the symbol is defined in.
+     * For example, in JS/TS this would be the npm module name. In Go, the full
+     * package import path. In PHP, the Composer package name. etc.
+     */
+    packageName?: string;
+
+    /**
+     * Whether or not the symbol is defined inside of "vendored" code. In Go, for
+     * example, this means that an external dependency was copied to a subdirectory
+     * named `vendor`. The exact definition of vendor depends on the language,
+     * but it is generally understood to mean "code that was copied from its
+     * original source and now lives in our project directly".
+     */
+    vendor?: boolean;
+}
+```
+
+### Goto Definition Extension Request
```

Comment:
@felixfbecker I added some explanation at the top of the file as part of this commit describing what it is used for, not sure if that wasn't enough or if you just missed it? I can add more information if that doesn't describe it well :)